### PR TITLE
Generic Checkout :  Signed-In

### PR DESCRIPTION
**Pull request by paul-daniel-dempsey**
## What are you doing in this PR?

Ensure Generic Checkout has a 'signed in' feature

image

[**Trello Card**](https://trello.com/c/EdqM43Oy/809-signed-in-on-generic-checkout)

### CODE REVIEW

File: support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx [hunk 1]
```diff
@@ -363,7 +363,8 @@ export function ThreeTierLanding(): JSX.Element {
 		),
 	);
 
-	const useGenericCheckout = abParticipations.useGenericCheckout === 'variant';
+	const useGenericCheckout =
+		abParticipations.useGenericCheckout.startsWith('variant');
```
Comment (jamesgorrie) [inline support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx:367]: Why this change?

**Review by jamesgorrie — APPROVED** [review]

File: support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx (hunk 1 above)
Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx:367]: The generic landing page with login url looked like this -> `https://support.code.dev-theguardian.com/uk/contribute#ab-useGenericCheckout=variant?INTCMP=mma_footer_support_contribute&...`. So I found `abParticipations.useGenericCheckout` was = `variant?INTCMP` not `variant` so I was forced to use `.startsWith('variant')` - maybe there is a better way?

Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx:368]: This has highlighted the need for a `!!abParticipations.useGenericCheckout && ` beforehand

Comment (jamesgorrie) [reply support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx:368]: This URL:
`https://support.code.dev-theguardian.com/uk/contribute#ab-useGenericCheckout=variant?INTCMP=mma_footer_support_contribute`

Will cause a load of issues. 

Mainly because the `window.location.hash` is now `ab-useGenericCheckout=variant?INTCMP=mma_footer_support_contribute`.

And the `window.location.seach` (QueryString) is now `undefined`.

I imagine the URL you're looking for is:
`https://support.code.dev-theguardian.com/uk/contribute?INTCMP=mma_footer_support_contribute#ab-useGenericCheckout=variant` (`hash` is always after the `search`).

Happy to pair to see how we got into this state.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx:368]: Ok your right, I was manually was testing/building the url incorrectly  myself with `.../contribute#ab-useGenericCheckout=variant?INTCMP=mma_footer_support_contribute...`. If its at the end yep works rfectly as is, will reset now.

Comment (paul-daniel-dempsey) [reply support-frontend/assets/pages/supporter-plus-landing/twoStepPages/threeTierLanding.tsx]: reverted
